public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +62,289 @@ const TITLE_TEXT_DICT = {
+}) {
+  return (
+    <FormRow
+      headerText="Document level security"
+      headerSubText="You can restrict a role to a subset of document in an index."
+      helpLink="/"
+      optional
+    >
+      <EuiTextArea
+        placeholder={FIELD_LEVEL_SECURITY_PLACEHOLDER}
+        value={props.value}
+        onChange={props.onChangeHandler}
+      />
+    </FormRow>
+  );
+}
+
+function FieldLevelSecurityRow(props: {
+  method: FieldLevelSecurityMethod;
+  fields: OptionSeletion;
+  onMethodChangeHandler: (s: string) => void;
+  onFieldChangeHandler: (s: OptionSeletion) => void;
+  onFieldCreateHandler: (s: string) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Field level security"
+      headerSubText="You can restrict what document field that user can see. If you use field level security in conjunction with document-level security, make sure you don't restrict to the field that document-level security uses."
+      optional
+    >
+      <EuiFlexGroup>
+        <EuiFlexItem grow={1}>
+          <EuiSuperSelect
+            valueOfSelected={props.method}
+            options={[
+              { inputDisplay: 'Include', value: 'include' },
+              { inputDisplay: 'Exclude', value: 'exclude' },
+            ]}
+            onChange={props.onMethodChangeHandler}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={9}>
+          <EuiComboBox
+            noSuggestions
+            placeholder="Type in field name"
+            selectedOptions={props.fields}
+            onChange={props.onFieldChangeHandler}
+            onCreateOption={props.onFieldCreateHandler}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </FormRow>
+  );
+}
+
+function AnonymizationRow(props: {
+  value: OptionSeletion;
+  onChangeHandler: (s: OptionSeletion) => void;
+  onCreateHandler: (s: string) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Anonymization"
+      headerSubText="Masks any sensitive fields with random value to protect your data security."
+      optional
+    >
+      <EuiComboBox
+        noSuggestions
+        placeholder="Type in field name"
+        selectedOptions={props.value}
+        onChange={props.onChangeHandler}
+        onCreateOption={props.onCreateHandler}
+      />
+    </FormRow>
+  );
+}
+
+function generateIndexPermissionPanels(
+  indexPermissions: RoleIndexPermissionStateClass[],
+  permisionOptionsSet: EuiComboBoxOptionOption[],
+  setRoleIndexPermission: Dispatch<SetStateAction<RoleIndexPermissionStateClass[]>>
+) {
+  const panels = indexPermissions.map((perm, index) => {
+    const onValueChangeHandler = (field: string) =>
+      updateElementInArrayHandler(setRoleIndexPermission, [index, field]);
+
+    const onCreateOptionHandler = (field: string) =>
+      appendOptionToComboBoxHandler(setRoleIndexPermission, [index, field]);
+
+    return (
+      <Fragment key={`index-permission-${index}`}>
+        <EuiAccordion
```

Comment:
Correct me if I am wrong. The same index can be used in two panels. Then there will be two Fragment with the same keys. Will this be a problem?

Reply:
Oh, I see. The word "index" here is really confusing. It is actually array index instead of that index in ES context. To make it easy to read, probably change the name from index to arrayIndex or indexInArray

Reply:
or positionInArray to avoid using the word index at all.

Reply:
Agreed.